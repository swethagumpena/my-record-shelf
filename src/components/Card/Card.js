import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import utilsApi from '../../utils/apiUtils';
import styles from './Card.module.scss';

const Card = ({
  id, albumArtUrl, songName, artistName,
}) => {
  const [like, setLike] = useState({});
  // like has like,count

  useEffect(async () => {
    try {
      const songLikes = await utilsApi.getLikes(id);
      setLike(songLikes);
    } catch (e) {
    //   setError(e);
    }
  }, []);

  return (
    <div className={styles.card}>
      <img data-testid="song-image" src={albumArtUrl} alt="song" />
      <div className={styles.content}>
        <div className={styles.text}>
          <p className={styles.song}>{songName}</p>
          <p className={styles.artist}>{artistName}</p>
        </div>
        <div className={styles.likeButton}>
          <button type="button">{like.count}</button>
        </div>
      </div>
    </div>
  );
};

Card.propTypes = {
  id: PropTypes.number.isRequired,
  albumArtUrl: PropTypes.string.isRequired,
  songName: PropTypes.string.isRequired,
  artistName: PropTypes.string.isRequired,
};

export default Card;
