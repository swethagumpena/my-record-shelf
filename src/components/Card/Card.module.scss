@import "../../styles/styles.scss";

.card {
    margin: 0.5rem;
    border-radius: 1.25rem;
    // box-shadow: 0rem 0rem 0.25rem #9a9a9a;
    // width: calc((100% - (8 * 0.125rem + 8 * 0.5rem)) / 4);
    width: 21rem;
    display: flex;
    flex-direction: column;
    border: 0.125rem solid lightgrey;
    align-items: flex-start;
    // background-color: $primary-background-color;
    // color: $primary-text-color;
    font-family: 'Poppins', sans-serif;

}

.card img{
    width: 70%;
    border-radius: 50%;
    padding: 1rem;
    align-self: center;
}

.song{
    margin: 1rem 1rem 0.5rem 1rem;
    font-size: 1.1rem;
}

.artist{
    margin: 0 1rem 2rem 1rem;
    font-weight: bolder;
    font-size: 0.8rem;
}

.content{
    display: flex;
    align-items: center;
}

.likeButton{
    float: left;
}

