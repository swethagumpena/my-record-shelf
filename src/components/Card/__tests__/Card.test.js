import { render, screen } from '@testing-library/react';
import React from 'react';
import Card from '../Card';

describe('example', () => {
  test('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
});

// const mockLikeCount = 2;

describe(Card.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(<Card id="1" albumArtUrl="mock" songName="mock" artistName="mock" />);
    expect(container).toMatchSnapshot();
  });

  test('should display song information', () => {
    render(<Card id="1" albumArtUrl="image" songName="song" artistName="artist" />);
    expect(screen.getByText('song'));
    expect(screen.getByText('artist'));
    screen.getByTestId('song-image');
  });

  test('should display button of  count', () => {
    render(<Card id="1" albumArtUrl="image" songName="song" artistName="artist" />);
    // const likeElement = screen.getByText({ mockLikeCount });
    // expect(likeElement.tagName).toBe('BUTTON');
    // fireEvent.click(likeElement);
  });
});
