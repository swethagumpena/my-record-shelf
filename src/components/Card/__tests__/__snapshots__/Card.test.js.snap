// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="song"
      data-testid="song-image"
      src="mock"
    />
    <div
      class="content"
    >
      <div
        class="text"
      >
        <p
          class="song"
        >
          mock
        </p>
        <p
          class="artist"
        >
          mock
        </p>
      </div>
      <div
        class="likeButton"
      >
        <button
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;
